lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -401,11 +401,12 @@ def bounce_service host, service, curl_retries = nil, port = nil
         # @param [Block] block This method will yield to a block of code passed
         #                      by the caller; this can be used for additional
         #                      validation, etc.
+        # @param [Boolean] run_in_parallel Whether to run on each host in parallel.
         #
         # @return [Array<Result>, Result, nil] An array of results, a result object,
         #   or nil. Check {#run_block_on} for more details on this.
-        def apply_manifest_on(host, manifest, opts = {}, &block)
-          block_on host do | host |
+        def apply_manifest_on(host, manifest, opts = {}, run_in_parallel=false, &block)
```

Comment:
Hm, but it's provided as an option in the `opts` elsewhere?  And it truly is an option.

I think we should keep it in the `options` hash, that way we won't need to modify a bunch of methods to accept a new parameter, and we can avoid lines that simply pull it out of the hash and pass it down explicitly; everything already accepts the `opts` hash and passes it around, we should use that.

> If you want to be able to execute the same method in parallel or not in parallel depending on the context, it pretty much needs to be outside of the options hash, unless I'm misunderstanding something

The location of the value shouldn't inhibit this; it can remain in the options hash, but would just need to be updated in their between method calls.

> any calls to these methods will still default to run_in_parallel=false

We can still do this without using method parameter default values, right?

> The reason for having this as a separate argument to the method is so you can opt on a per call basis whether to execute in parallel or not.

That's still true when it's an `options`, it just takes a few extra characters of code, right?

Overall, I think it would be wise to organize this based on the structure of the code and not on the amount of effort/characters necessary to use it.  Since we've already got `options` plumbed throughout the code, and we're introducing a new option, then it would make sense to fold it into the existing convention.  It does require a tiny bit more code, but that's pretty cheap.